### DISCUSSION TITLE
scp custom command fails

### DISCUSSION THREAD

**Discussion by daldr-ntml** [microsoft/vscode-cmake-tools#2925] [Q&A] [answered]
Hi, my CMakeLists.txt has a post build custom command to copy the target to a remote server using scp:

`add_custom_command(TARGET ${EXECUTABLE}
                           POST_BUILD
                           COMMAND scp ${EXECUTABLE}.vhx $ENV{USERNAME}@myserver:/home/$ENV{USERNAME}/<snip>/cm4.vhx)`

This works when I build from the command line, but when building from VS Code I see:
```
[build] process_begin: CreateProcess(NULL, scp arm.vhx <snip>@myserver:<snip>/cm4.vhx, ...) failed.
[build] make (e=2): The system cannot find the file specified.
[build] make.exe[2]: *** [CMakeFiles/<snip>/build.make:615: elysion_arm.elf] Error 2
[build] make.exe[2]: Leaving directory 'C:/<snip>/builds/modelsim-release'
```
Does this suggest that the build process can't find 'scp'? 

Any suggestions on how to debug/fix this would be appreciated.

**Comment by bobbrow** [comment, marked as answer]
This looks like it can't find `scp`.  You can try removing the arguments to double check, but I just did a quick test with `echo` vs `echo2` and it works for the first one and fails for the second.

**Reply by daldr-ntml** [reply]
Thank you Bob, my path was incorrect.
